pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -571,10 +583,22 @@ func (p *staticPolicy) generateCPUTopologyHints(availableCPUs cpuset.CPUSet, reu
 	// to the minAffinitySize. Only those with an equal number of bits set (and
 	// with a minimal set of numa nodes) will be considered preferred.
 	for i := range hints {
+		if p.options.AlignBySocket && p.isHintSocketAligned(hints[i].NUMANodeAffinity) {
+			hints[i].Preferred = true
+			continue
+		}
 		if hints[i].NUMANodeAffinity.Count() == minAffinitySize {
 			hints[i].Preferred = true
 		}
 	}
 
 	return hints
 }
+
+func (p *staticPolicy) isHintSocketAligned(hint bitmask.BitMask) bool {
+	numaNodes := hint.GetBits()
+	if p.topology.CPUDetails.SocketsInNUMANodes(numaNodes[:]...).Size() == 1 {
+		return true
+	}
+	return false
```

Comment:
nit: ```return p.topology.CPUDetails.SocketsInNUMANodes(numaNodes[:]...).Size() == 1```
(I'd not resubmit _just_ for this)

Reply:
Noted! Will take care in subsequent commit